File: xenxin/GWA2/java/WEB-INF/classes/com/ufqi/gwa2/inc/WebAppInterface.java
/*
 *
 * GWA2 in useBean with .java
 * import from GWA2 in .jsp
 * wadelau@{ufqi, hotmail, gmail}.com
 * 
 * since Tue Apr  9 03:28:28 UTC 2019
 */

package com.ufqi.gwa2.inc;

import java.util.HashMap;

public interface WebAppInterface{

    public void set(String key, Object value);
    public String get(String key);

    public HashMap setBy(String fields, String conditions);
    public HashMap getBy(String fields, String conditions);

    //public void setTbl(String tbl);
    //public String getTbl();

    public void setId(String iId);
    public String getId();

    public HashMap execBy(String fields, String conditions);
    public HashMap rmBy(String conditions);

    /*
       public String toString(Object obj);
     */

}

File: xenxin/GWA2/php/inc/webapp.interface.php
<?php
/* WebApp Interface definition for all of the implement classes
 * v0.1,
 * wadelau@ufqi.com, 2011-07-10 15:27
 * remedy by wadelau@ufqi.com, 10:12 01 May 2016
 */

interface WebAppInterface{
	
	function set($key, $value);
	function get($key);

	function setTbl($tbl);
	function getTbl();
		
	function setId($id);
	function getId();

	function setBy($fields, $conditions);
	function getBy($fields, $conditions);
	function execBy($fields, $conditions);
	function rmBy($conditions);

  	function toString($object);
	
}
?>
